pkg/services/hcloud/loadbalancer/loadbalancer.go

```diff
@@ -358,42 +343,39 @@ func (s *Service) findLoadBalancer(ctx context.Context) (*hcloud.LoadBalancer, e
 	return loadBalancers[0], nil
 }
 
-// gets the information of the Hetzner load balancer object and returns it in our status object.
-func apiToStatus(lb *hcloud.LoadBalancer, hasNetwork bool) (infrav1.LoadBalancerStatus, error) {
-	ipv4 := lb.PublicNet.IPv4.IP.String()
-	ipv6 := lb.PublicNet.IPv6.IP.String()
-
+// statusFromHCloudLB gets the information of the Hetzner load balancer and returns it in the status object.
+func statusFromHCloudLB(lb *hcloud.LoadBalancer, hasNetwork bool) *infrav1.LoadBalancerStatus {
 	var internalIP string
 	if hasNetwork && len(lb.PrivateNet) > 0 {
 		internalIP = lb.PrivateNet[0].IP.String()
 	}
 
-	targets := make([]infrav1.LoadBalancerTarget, 0, len(lb.Targets))
+	targetObjects := make([]infrav1.LoadBalancerTarget, 0, len(lb.Targets))
 	for _, target := range lb.Targets {
 		switch target.Type {
 		case hcloud.LoadBalancerTargetTypeServer:
-			targets = append(targets, infrav1.LoadBalancerTarget{
+			targetObjects = append(targetObjects, infrav1.LoadBalancerTarget{
 				Type:     infrav1.LoadBalancerTargetTypeServer,
 				ServerID: target.Server.Server.ID,
 			},
 			)
 		case hcloud.LoadBalancerTargetTypeIP:
-			targets = append(targets, infrav1.LoadBalancerTarget{
+			targetObjects = append(targetObjects, infrav1.LoadBalancerTarget{
 				Type: infrav1.LoadBalancerTargetTypeIP,
 				IP:   target.IP.IP,
 			},
 			)
 		default:
-			return infrav1.LoadBalancerStatus{}, fmt.Errorf("unknown load balancer target type %s", target.Type)
+			panic(fmt.Sprintf("unknown load balancer target type %s", target.Type))
```

Comment:
Are there alternatives to panic?

Maybe create an event?

Reply:
Events would create an overhead here. The panic should never come up, because this is something that won't change immediately. It would make it obvious though that there was a big change.
I guess the alternative would be for me to just not say anything and just not add the load balancer targets with unknown type. 
What do you think?

Reply:
ok, logging an error and setting the value to same sane default value would be good. I would avoid to panic.